src/test/kotlin/uk/gov/dluhc/printapi/service/CertificateRemovalDateResolverTest.kt

```diff
@@ -26,40 +29,67 @@ internal class CertificateRemovalDateResolverTest {
+            val expectedRemovalDate = LocalDate.of(2023, 2, 9)
+            given(dataRetentionConfig.certificateInitialRetentionPeriod).willReturn(Period.ofDays(28))
+            given(bankHolidayDataClient.getBankHolidayDates(any(), any(), any())).willReturn(listOf(upcomingBankHoliday))
 
-        // When
-        val actual = certificateRemovalDateResolver.getCertificateInitialRetentionPeriodRemovalDate(issueDate, gssCode)
+            // When
+            val actual = certificateRemovalDateResolver.getCertificateInitialRetentionPeriodRemovalDate(issueDate, gssCode)
 
-        // Then
-        assertThat(actual).isEqualTo(expectedRemovalDate)
-        verify(dataRetentionConfig).certificateInitialRetentionPeriod
-        verify(bankHolidayDataClient).getBankHolidayDates(ENGLAND_AND_WALES)
+            // Then
+            assertThat(actual).isEqualTo(expectedRemovalDate)
+            verify(dataRetentionConfig).certificateInitialRetentionPeriod
+            verify(bankHolidayDataClient).getBankHolidayDates(ENGLAND_AND_WALES)
+        }
+
+        @Test
+        fun `should get initial retention period removal date for delivery info given no upcoming bank holidays`() {
+            // Given
+            val gssCode = "E09000007"
+            val issueDate = LocalDate.of(2023, 1, 1)
+            val expectedRemovalDate = LocalDate.of(2023, 2, 8)
+            given(dataRetentionConfig.certificateInitialRetentionPeriod).willReturn(Period.ofDays(28))
+            given(bankHolidayDataClient.getBankHolidayDates(any(), any(), any())).willReturn(emptyList())
+
+            // When
+            val actual = certificateRemovalDateResolver.getCertificateInitialRetentionPeriodRemovalDate(issueDate, gssCode)
+
+            // Then
+            assertThat(actual).isEqualTo(expectedRemovalDate)
+            verify(dataRetentionConfig).certificateInitialRetentionPeriod
+            verify(bankHolidayDataClient).getBankHolidayDates(ENGLAND_AND_WALES)
+        }
     }
 
-    @Test
-    fun `should get removal date for delivery info given no upcoming bank holidays`() {
-        // Given
-        val gssCode = "E09000007"
-        val issueDate = LocalDate.of(2023, 1, 1)
-        val expectedRemovalDate = LocalDate.of(2023, 2, 8)
-        given(dataRetentionConfig.certificateInitialRetentionPeriod).willReturn(Period.ofDays(28))
-        given(bankHolidayDataClient.getBankHolidayDates(any(), any(), any())).willReturn(emptyList())
+    @Nested
+    inner class ElectorDocumentFinalRetentionPeriod {
+        @ParameterizedTest
+        @CsvSource(
+            "2023-01-01, 2032-07-01",
+            "2023-06-30, 2032-07-01",
+            "2023-07-01, 2033-07-01",
+            "2023-07-02, 2033-07-01",
+            "2024-01-01, 2033-07-01",
+            "2024-12-31, 2034-07-01"
+        )
+        fun `should get final retention period removal date for elector document`(issueDateStr: String, targetDateStr: String) {
```

Comment:
minor but junit will automatically coerce strings to LocalDates i.e. 
```
fun `should get final retention period removal date for elector document`(issueDate: LocalDate, expectedTargetDate: LocalDate) {
```

Reply:
Ah yes, good point. I meant to come back to that and check, but forgot! 😄